File: client/src/types/tunnel.ts
export interface TunnelDTO {
    id: string
    localPort: number
    remotePort: number
    protocol: 'tcp' | 'udp' | 'http' | 'https'
    status: 'active' | 'inactive' | 'closed'
    createdAt: string
}

export interface CreateTunnelDTO {
    localPort: number
    remotePort: number
    protocol: string
}

File: crates/server/src/domain/models/tunnel.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Tunnel {
    pub id: Uuid,
    pub client_id: Uuid,
    pub local_port: u16,
    pub remote_port: u16,
    pub protocol: TunnelProtocol,
    pub status: TunnelStatus,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum TunnelProtocol {
    Tcp,
    Udp,
    Http,
    Https,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum TunnelStatus {
    Active,
    Inactive,
    Closed,
}
